test/tests/dirackernels/stateful_point_source/tests

```diff
@@ -1,7 +1,8 @@
 [Tests]
   [./stateful_point_source_test]
-    type = 'RunException'
+    type = 'Exodiff'
     input = 'stateful_point_source.i'
-    expect_err = "Error: Stateful material properties not allowed"
+    exodiff = "out.e"
+    recover = false
```

Comment:
I was looking into the recover issue and I think it fails because the material property storage is not initialized before the restart starts to run. This comes back to the basic design question, such as when should we init/reinit/resize material property value for these ```Dirac``` points. If there are any good solution, I would like to give it a shot. :-)